### ISSUE TITLE
Unable to find headers & autoreconfig not found and 

### ISSUE THREAD

**Issue by midair55** [Allan-N/ASL-Install#9]
Step 1 - Setup errors with:  

E: Unable to locate package linux-headers-6.1.21-v7
E: Couldn't find any package by glob 'linux-headers-6.1.21-v7'

Raspberry Pi 3B+ running Debian Bookworm:

$ uname -r
6.1.21-v7+

<><><>
Step 3 - Errors with autoreconfig not found.   This was resolved using, "sudo apt-get install autoconf" however Step 3 errored during the compile with the following found in /var/tmp/clean-dahdi.txt

* DAHDI
make: Entering directory '/home/pi/AllStarLink/ASL-Install/ASL-DAHDI'
make -C linux clean
make[1]: Entering directory '/home/pi/AllStarLink/ASL-Install/ASL-DAHDI/linux'
make -C drivers/dahdi/firmware clean
make[2]: Entering directory '/home/pi/AllStarLink/ASL-Install/ASL-DAHDI/linux/drivers/dahdi/firmware'
rm -f dahdi-fw-*.o
make[2]: Leaving directory '/home/pi/AllStarLink/ASL-Install/ASL-DAHDI/linux/drivers/dahdi/firmware'
make -C  M='/home/pi/AllStarLink/ASL-Install/ASL-DAHDI/linux/drivers/dahdi/oct612x' clean
make[2]: Entering directory '/home/pi/AllStarLink/ASL-Install/ASL-DAHDI/linux'
make[2]: *** M=/home/pi/AllStarLink/ASL-Install/ASL-DAHDI/linux/drivers/dahdi/oct612x: No such file or directory.  Stop.
make[2]: Leaving directory '/home/pi/AllStarLink/ASL-Install/ASL-DAHDI/linux'
make[1]: *** [Makefile:182: clean] Error 2
make[1]: Leaving directory '/home/pi/AllStarLink/ASL-Install/ASL-DAHDI/linux'
make: *** [Makefile:14: clean] Error 2
make: Leaving directory '/home/pi/AllStarLink/ASL-Install/ASL-DAHDI'

**Comment by Allan-N** [comment]
I'm tracking another issue where an "apt upgrade" that included bumping the kernel was not followed by a reboot and we ended up with a header mismatch.

Can you reboot your device, re-exec the `asl-install` script, and start the build/install over with step 1, 2, 3, etc.  Does the install work this time?

**Comment by midair55** [comment]
Allan:

Thanks for the quick response!!!!    I do the following every Sunday morning via a crontab script:

/usr/bin/sudo apt-get update -y
/usr/bin/sudo apt-get upgrade -y
/usr/bin/sudo apt-get install -y
/usr/bin/sudo apt autoremove -y
/usr/bin/sudo reboot

<><><>

I manually rebooted as you requested and it failed to find the headers as before.  I also power cycled the Pi and it continues not to find the headers.

Steve - N5NG
________________________________

**Comment by Allan-N** [comment]
The DAHDI project is compiled for a specific version of the kernel so one needs to be careful about upgrades (upgrade the kernel and then you must rebuild DAHDI).

As to why you are having issues, I'm at a loss.

The `asl-install` script's first step ("1. Setup") tries to install the linux-headers for the currently running kernel.  Is it possible that you exec'd a command to block updates to some of the apt repo's?  Do you see any holds when exec'ing `sudo apt-mark showhold`?  If so, does removing the hold help you out?

What kernel version do you have installed ... `uname -srv` ?

Which linux-header packages to you have installed ... `sudo apt list --installed | grep linux-headers` ?

If you have any linux-header packages that do not match the running kernel then you certainly try removing them.

**Comment by midair55** [comment]
response below in red.

________________________________

**Comment by Allan-N** [comment]
> I'll burn a new Bookworm OS to an SD and boot up and follow the steps on github to see if having a clean(er) OS helps.

Please reboot your device after you exec `sudo apt upgrade` and before you exec `asl-install`

**Comment by Allan-N** [comment]
Also, if you run into trouble, it would be good to know what is reported when trying to just install the headers :

```
sudo apt install -y linux-headers-`uname -r`
```

**Comment by midair55** [comment]
roger

**Comment by midair55** [comment]
Fresh Debian Bullseye 32-bit using Raspberry Pi Imager to flash the SD.   Before any 'update/upgrade':

 $ uname -srv
Linux 6.1.21-v7+ #1642 SMP Mon Apr  3 17:20:52 BST 2023

Same is seen after the 'update/upgrade'.

Rebooted after the 'update/upgrade' since issues have been seen where this was not done....

Executed:  asl-install.sh as a non-root user....   Step 1 errored as before:

E: Unable to locate package linux-headers-6.1.21-v7
E: Couldn't find any package by glob 'linux-headers-6.1.21-v7'

<><>

Exited from the asl-install.sh after step 1 and confirmed to reboot.

Executed asl-install.sh, step 1.  Same error.

Note:  I'm using Bullseye.  I'll try Bookworm.




________________________________

**Comment by Allan-N** [comment]
Oh, yes, please try Bookworm.

**Comment by midair55** [comment]
Allan - Sorry for the slow response.....   I was successful installing on the Pi 3B+ using Bookworm.  It appears Bullseye is a bad player....   Thanks for all of your help and effort in this development.  It simplifies the install tremendously!!!
Cheers!
Steve -

**Comment by midair55** [comment]
issue closed
